File: Projects/leetcode/MaxSubArray.java
package leetcode;

class MaxSubArray {
    public static int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int curSum = 0;
        for (int num : nums) {
            curSum = curSum >= 0 ? curSum + num : num;
            maxSum = Math.max(curSum, maxSum);
        }
        return maxSum;
    }
}

File: Projects/leetcode/maxSubArray.kt
package leetcode

fun maxSubArray(nums: IntArray): Int {
    var maxSum = Int.MIN_VALUE
    var curSum = 0
    for (num in nums) {
        curSum = if (curSum >= 0) curSum + num else num
        maxSum = Math.max(curSum, maxSum)
    }
    return maxSum
}
